password_security/models/res_users.py

```diff
@@ -72,6 +76,31 @@ def password_match_message(self):
     def _check_password(self, password):
         self._check_password_rules(password)
         self._check_password_history(password)
+        self._check_password_similar_name(password, self.name)
+        self._check_password_similar_login(password, self.login)
+        return True
+
+    @api.multi
+    def _check_password_similar_name(self, password, name):
+        self.ensure_one()
+        if self.company_id.password_not_similar_name_login:
+            # remove non-alnums from user's name for comparison
+            name_alnum = re.sub(r'[\W_]', '', name)
+            if name_alnum.lower() == password.lower():
```

Comment:
But that is checking equality, not similarity 🤔

Reply:
This is unneededly overly complex IMHO. Simpler:

```python
password = password.lower()
name = name.lower()
if password in name or name in password:
    raise PassError(self.password_match_message())
```